### PR TITLE
Replace Data Warehouse Export S3 upload with Azure

### PR DESCRIPTION
**Pull request by james**
WARNING: Do not merge before setting `AZURE_ACCOUNT_NAME`, `AZURE_ACCOUNT_KEY` and `AZURE_CONTAINER_NAME` in the ENV on staging and production.

As part of our transition to GPAAS, we have realised it would be much more beneficial to store the export files in CCS's Azure, rather than S3.

### CODE REVIEW

File: spec/models/export/azure_upload_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  let(:files_map) { { 'file1.csv' => file_1, 'file2.json' => file_2 } }
+  let(:file_1) { File.open(Rails.root.join('spec', 'fixtures', 'users.csv')) }
+  let(:file_2) { File.open(Rails.root.join('spec', 'fixtures', 'bank_holidays.json')) }
+
+  subject(:azure_upload) { Export::AzureUpload.new(files_map) }
+
+  let(:blob_client) { spy('blob_client') }
+
+  around do |example|
+    ClimateControl.modify AWS_S3_EXPORT_BUCKET: 'test-bucket' do
```
Comment (leeky) [inline spec/models/export/azure_upload_spec.rb]: Should this be `AZURE_CONTAINER_NAME` ..?

File: spec/models/export/azure_upload_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      example.run
+    end
+  end
+
+  describe '#perform' do
+    it 'uploads each file to the configured S3 bucket' do
```
Comment (leeky) [inline spec/models/export/azure_upload_spec.rb]: `it 'uploads each file to the configured Azure container' do`

File: spec/models/export/azure_upload_spec.rb (hunk 1 above)
Comment (james) [reply spec/models/export/azure_upload_spec.rb]: Doh! Yes, it should. Which has made me realise I'm not actually dynamically setting the container name 🤦‍♂️

File: spec/models/export/azure_upload_spec.rb (hunk 2 above)
Comment (james) [reply spec/models/export/azure_upload_spec.rb]: 🤦‍♂️

**Comment by rjw1** [comment]
Could you do a pull request to add the environment variables to https://github.com/dxw/DataSubmissionService/blob/develop/CF/user-services.json and https://github.com/dxw/DataSubmissionService/blob/develop/CF/env-cf-example

Also in this PR add the service name to https://github.com/dxw/DataSubmissionServiceAPI/blob/develop/CF/manifest-template.yml

**Comment by james** [comment]
> Could you do a pull request to add the environment variables to https://github.com/dxw/DataSubmissionService/blob/develop/CF/user-services.json and https://github.com/dxw/DataSubmissionService/blob/develop/CF/env-cf-example

@rjw1 Did you mean that to be on the supplier facing app, or on API?

**Comment by james** [comment]
@rjw1 Ah, I see, I think we're storing the user services for both apps in supplier facing app. Made a PR, is that right? https://github.com/dxw/DataSubmissionService/pull/233

**Pull request merged**